script/setup/common.ts

```diff
@@ -15,15 +15,15 @@ export type ProjectConfig = {
   hookType: Hooks;
   owner: string;
   isMainnet: boolean;
-  newToken: boolean;
+  newToken?: boolean;
 };
 
 export type NewTokenInfo = {
-  name: string;
-  symbol: string;
-  decimals: number;
-  chainSlug: ChainSlug;
   address: string;
+  name?: string;
```

Comment:
why are we making these optional? want to understand the flow better

Reply:
I had before a interface where you would either supply an address, then nothing else, or everything else but not the address. To define whether the token needs to be deployed or is already deployed.